src/databricks/labs/ucx/progress/workflows.py

```diff
@@ -22,26 +22,34 @@ class MigrationProgress(Workflow):
     def __init__(self) -> None:
         super().__init__('migration-progress-experimental')
 
-    @job_task
+    @job_task(job_cluster="table_migration")
+    def verify_prerequisites(self, ctx: RuntimeContext) -> None:
+        """Verify the prerequisites for running this job on the table migration cluster are fulfilled.
+
+        We will wait up to 1 hour for the assessment run to finish if it is running or pending.
+        """
+        ctx.verify_progress_tracking.verify(timeout=dt.timedelta(hours=1))
+
+    @job_task(depends_on=[verify_prerequisites], job_cluster="table_migration")
     def crawl_tables(self, ctx: RuntimeContext) -> None:
         """Iterates over all tables in the Hive Metastore of the current workspace and persists their metadata, such
         as _database name_, _table name_, _table type_, _table location_, etc., in the table named
         `$inventory_database.tables`. The metadata stored is then used in the subsequent tasks and workflows to, for
         example, find all Hive Metastore tables that cannot easily be migrated to Unity Catalog."""
-        ctx.tables_crawler.snapshot(force_refresh=True)
+        history_log = ctx.historical_tables_log
+        tables_snapshot = ctx.tables_crawler.snapshot(force_refresh=True)
+        history_log.append_inventory_snapshot(tables_snapshot)
 
-    @job_task
+    @job_task(depends_on=[verify_prerequisites], job_cluster="table_migration")
     def crawl_udfs(self, ctx: RuntimeContext) -> None:
         """Iterates over all UDFs in the Hive Metastore of the current workspace and persists their metadata in the
         table named `$inventory_database.udfs`. This inventory is currently used when scanning securable objects for
         issues with grants that cannot be migrated to Unit Catalog."""
-        ctx.udfs_crawler.snapshot(force_refresh=True)
-
-    @job_task(job_cluster="tacl")
-    def setup_tacl(self, ctx: RuntimeContext) -> None:
-        """(Optimization) Starts `tacl` job cluster in parallel to crawling tables."""
+        history_log = ctx.historical_udfs_log
+        udfs_snapshot = ctx.udfs_crawler.snapshot(force_refresh=True)
+        history_log.append_inventory_snapshot(udfs_snapshot)
 
-    @job_task(depends_on=[crawl_tables, crawl_udfs], job_cluster="tacl")
+    @job_task(depends_on=[verify_prerequisites, crawl_tables, crawl_udfs], job_cluster="table_migration")
```

Comment:
did you verify that `table_migration` can crawl legacy TACL?

Reply:
Indeed, I was testing this (it's covered by integration tests).

It indeed fails, because the table-migration cluster is using Spark Connect and the fast crawler doesn't have the JVM-bridge into Spark available.

I've added a note on `RuntimeContext.tables_crawler` that as things stand not all runtime contexts support crawling.

Reply:
So this created a bit of a problem, for which a workaround is now in place:

 - The TACL cluster, where we can crawl from, is not UC-enabled. So we can't write to the history log.
 - The table-migration cluster, where can write to the log, can't crawl tables.

The workaround for now is to split the table-refresh into 2 tasks:
1. Crawl from the `tacl` cluster, and just update the (HMS-based) inventory.
2. On the `table-migration` cluster, read the current inventory snapshot (assumably from step 1) and update the history log.

It's not perfect, in particular because if crawling produced no records (for whatever reason) then a crawl will be triggered during step 2.